manifests/testing-framework/testing-framework.yaml

```diff
@@ -30,6 +30,11 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
+            - name: AUTO_CLEAN_UP
```

Comment:
Hi @MiroslavRepka , I'm just wondering whether this works when AUTO_CLEAN_UP is missing from the `env` file, e.g. in cases where you try to deploy it manually.
But I guess that's not a big deal anyway.

Reply:
hey @MarioUhrik, if the question is whether the infra would get auto-cleaned up when `AUTO_CLEAN_UP` variable is not present in the environment, the answer is no.

the current implementation [performs a lookup](https://github.com/Berops/claudie/blob/5ef343dcc2f735ecb76db78d64334ded11e1b055/services/testing-framework/claudie_test.go#L40) using [`os.Getenv`](https://pkg.go.dev/os#Getenv), which doesn't really mind if the ENV var is not present; it simply returns an empty string in that case, which subsequently fails our check `if ... == "TRUE"` [here](https://github.com/Berops/claudie/blob/5ef343dcc2f735ecb76db78d64334ded11e1b055/services/testing-framework/claudie_test.go#L182), i.e. the infrastructure would not get automatically cleaned up.

Somebody correct me, please, if I'm wrong.

Reply:
I was specifically wondering whether Kubernetes can accept a Deployment with an env variable sourced from a configmap where such a variable isn't present

Reply:
ah, I see, that's a good question.

the [Optional ConfigMaps](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#optional-configmaps) section ([added](https://github.com/kubernetes/website/pull/35160) not so long ago) should answer the original question: "If the ConfigMap exists, but the referenced key is non-existent the data is also empty.".

that is, the `AUTO_CLEAN_UP` key needs to be marked as **optional**.